packages/runtime/container-runtime/src/dataStoreContext.ts

```diff
@@ -310,6 +313,23 @@ export abstract class FluidDataStoreContext extends TypedEventEmitter<IFluidData
         }
     }
 
+    public async tombstone() {
+        this.verifyNotClosed();
+        if (this.tombstoned) {
+            return;
+        }
+
+        assert(!(await this.isRoot()), "Tombstone should not be called on root datastores!");
+        assert(!this.isLocalDataStore, "Local data should have been wiped before tombstone is called!");
```

Comment:
Can local datastores last long after they are gone from the cache? It was just a thought/question I had, I left this assert in for that discussion. If not, then we wouldn't want to have this assert here.

Reply:
The thought here was that the context of a deleted datastore should be a remote datastore since the timers should have ensured that local data was not persisted. If a local datastore was created, then it would be from memory, and that did not get cleared, we would want a way to detect that.

Reply:
I am not sure I follow this. What has local data store got to do with cache? I also did not understand what you mean local data store not cleared from memory or cache.

Reply:
What are the cases for the creation of a `LocalFluidDataStoreContextBase` vs a `RemoteFluidDataStoreContext`?